docs/reference/scripting/using.asciidoc

```diff
@@ -1,425 +1,448 @@
+the `painless_test` context by default. This context optionally makes
+available a `Map` of user-defined parameters, and then returns a value as an
+`Object`.
 
-For more information, see <<security-privileges>>.
+The `script` is a standard JSON object that defines scripts under most APIs
+in {es}. This object requires `source` to define the script itself. The
+example script doesn't specify a language, so it uses Painless by default.
 
+The `source` defines the script itself. The example `source` returns a
+`String` value of `Hello, world!`, converts the string to an `Object`, and
+displays the output.
 
 [discrete]
-==== Request examples
-
-The following are examples of using a stored script that lives at
-`/_scripts/{id}`.
-
-First, create the script called `calculate-score` in the cluster state:
+[[script-add-parameters]]
+=== Add parameters to your script
+Now that you have a simple script, we'll introduce parameters to pass in
+variables. Using named `params` instead of hard-coding values makes scripts
+flexible, and also reduces compilation time when the script runs.
 
 [source,console]
------------------------------------
-POST _scripts/calculate-score
+----
+POST /_scripts/painless/_execute
 {
   "script": {
-    "lang": "painless",
-    "source": "Math.log(_score * 2) + params.my_modifier"
+    "source": "return params.get('hello_world');", <1>
+    "lang": "painless", <2>
+    "params": { <3>
+        "hello_world": "Hello, world!"
+    }
   }
 }
------------------------------------
-// TEST[setup:my_index]
-
-You may also specify a context as part of the url path to compile a
-stored script against that specific context in the form of
-`/_scripts/{id}/{context}`:
+----
+<1> Script `source`
+<2> Language declaration
+<3> `Map` of user-defined parameters
 
-[source,console]
------------------------------------
-POST _scripts/calculate-score/score
-{
-  "script": {
-    "lang": "painless",
-    "source": "Math.log(_score * 2) + params.my_modifier"
-  }
-}
------------------------------------
-// TEST[setup:my_index]
+Our script now contains a `Map` of user-defined parameters. The `get` method
+accesses a value using the `String` key of the `hello_world` parameter.
 
-This same script can be retrieved with:
+We'll explicitly specify the `painless` language, even though that's the
+default language.
 
-[source,console]
------------------------------------
-GET _scripts/calculate-score
------------------------------------
-// TEST[continued]
+The `params` section is where things get interesting. This part of the script
+is where you pass in user-defined parameters, which are written as a JSON
+object that is parsed into a `Map`. Our script defines a key of `hello_world`
+with a `String` value of `Hello, world!`. We then access this value within
+the example script.
 
-Stored scripts can be used by specifying the `id` parameters as follows:
+[discrete]
+[[script-shorten-syntax]]
```

Comment:
Let's keep the old version of this section, it's practical.  However it can be enhanced to describe the use case and context.

> For example, to [update](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update.html) a doc using a [script](https://www.elastic.co/guide/en/elasticsearch/painless/master/painless-update-context.html), you can use a shortened form of the request:
```
POST <index>/_update/<doc_id>
{
   "script" : "ctx['_source'].number++"
}
```

Rather than the long form:
```
POST <index>/_update/<doc_id>
{
   "script": {
     "lang": "painless",
     "source": "ctx['_source'].number++"
   }
}
```

Reply:
I like this direction, but think that the examples feel a bit discombobulated. Instead, I think that incorporating some of the examples from the https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update.html#update-api-example[Update API] page would be really useful.